subsys/dfu/dfu_target/src/dfu_target_modem_delta.c

```diff
@@ -28,68 +23,47 @@ struct modem_delta_header {
 	uint32_t magic;
 };
 
-static int  fd;
-static int  offset;
+static int offset;
 static dfu_target_callback_t callback;
 
-static int get_modem_error(void)
-{
-	int rc;
-	int err = 0;
-	socklen_t len;
-
-	len = sizeof(err);
-	rc = getsockopt(fd, SOL_DFU, SO_DFU_ERROR, &err, &len);
-	if (rc) {
-		LOG_ERR("Unable to fetch modem error, errno %d", errno);
-	}
-
-	return err;
-}
-
 static int apply_modem_delta_upgrade(void)
 {
 	int err;
 
 	LOG_INF("Scheduling modem firmware upgrade at next boot");
 
-	err = setsockopt(fd, SOL_DFU, SO_DFU_APPLY, NULL, 0);
-	if (err < 0) {
-		if (errno == ENOEXEC) {
-			LOG_ERR("SO_DFU_APPLY failed, modem error %d",
-				get_modem_error());
-		} else {
-			LOG_ERR("SO_DFU_APPLY failed, modem error %d", err);
-		}
+	err = nrf_modem_delta_dfu_update();
+	if (err != 0) {
+		LOG_ERR("Modem firmware upgrade scheduling failed, error %d", err);
 	}
```

Comment:
I'd move this into `dfu_target_modem_delta_done`; that function does not do anything more than calling this one.